src/main/java/com/example/Library/controller/BooksController.java

```diff
@@ -27,8 +30,10 @@ public BooksController(BookService bookService) {
 
     @PreAuthorize("hasAuthority('user:guest')")
     @GetMapping()
-    public ArrayList<Book> getAllBook(@RequestParam(value = "title",required = false)String title) {
-        return (ArrayList<Book>) bookService.getAllBooksWithFilterByTitle(title);
+    public Page<Book> getAllBook(@RequestParam (value = "title", required = false)String title,
+                                 @RequestParam (value = "page", required = false)Optional<Integer> page,
```

Comment:
pls, double-check if the usage of 'required = false' parameter is needed when Optional argument is used. If not - remove 'required' argument